pkg/util/util.go

```diff
@@ -249,3 +258,26 @@ func OpenBrowser(url string) error {
 
 	return nil
 }
+
+// FetchResourceQunatity takes passed min, max and requested resource quantities and returns min and max resource requests
+func FetchResourceQunatity(resourceType corev1.ResourceName, min string, max string, request string) ResourceRequirementInfo {
+	// If minimum and maximum both are passed they carry highest priority
+	// Otherwise, use the request as min and max
+	var minResource resource.Quantity
+	var maxResource resource.Quantity
+	if min != "" {
+		minResource = resource.MustParse(min)
```

Comment:
Would it be a good idea to return an error here and use `odoutil.CheckError(err, "")` in the `create.go`?

Reply:
Ok I think you meant returning an error if a wrong param is passed(vs an error @line 269), in which case the underlying lib raises a panic ... and hence odo would then fail

Regarding, ensuring that atleast one of the flags is set without which, we don't venture into setting this config in the dc, I'll add code handling it

Reply:
> Ok I think you meant returning an error if a wrong param is passed like an incorrect unit(vs an error @line 269), in which case the underlying lib raises a panic ... and hence odo would then fail
> 
> Regarding, ensuring that atleast one of the flags is set without which, we don't venture into setting this config in the dc, I'll add code handling it

I just tried to verify the case of wrong units it seems the pod creation fails but not in odo layer :(
@kadel Should we add custom validation of units ?
Like if I pass `--memory 300` instead of `--memory 300Mi` the vendored package("k8s.io/apimachinery/pkg/api/resource") doesn't seem to return error or panic as the func doc suggests

Reply:
This fails although:

```
[redhat@localhost odo]$ odo create nodejs --local ~/git-srcs/src/github.com/redhat-developer/odo-samples/nodejs-ex -v 4 --min-memory 200uu
I1119 19:22:31.134385    3083 create.go:76] Component create called with args: "nodejs", flags: binary=, git=, local=/home/redhat/git-srcs/src/github.com/redhat-developer/odo-samples/nodejs-ex
I1119 19:22:31.137483    3083 occlient.go:365] Trying to connect to server 192.168.42.137:8443
I1119 19:22:31.137784    3083 occlient.go:371] Server https://192.168.42.137:8443 is up
I1119 19:22:31.171986    3083 occlient.go:294] isLoggedIn err:  <nil> 
 output: "developer"
I1119 19:22:31.172229    3083 context.go:202] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: component
I1119 19:22:31.237277    3083 catalog.go:161] Found builder images: [{dotnet openshift [2.0 latest]} {httpd openshift [2.4 latest]} {nginx openshift [1.10 1.12 1.8 latest]} {nodejs openshift [0.10 10 4 6 8 8-RHOAR latest]} {perl openshift [5.16 5.20 5.24 5.26 latest]} {php openshift [5.5 5.6 7.0 7.1 latest]} {python openshift [2.7 3.3 3.4 3.5 3.6 latest]} {ruby openshift [2.0 2.2 2.3 2.4 2.5 latest]} {wildfly openshift [10.0 10.1 11.0 12.0 13.0 8.1 9.0 latest]}]
I1119 19:22:31.258837    3083 catalog.go:161] Found builder images: [{dotnet openshift [2.0 latest]} {httpd openshift [2.4 latest]} {nginx openshift [1.10 1.12 1.8 latest]} {nodejs openshift [0.10 10 4 6 8 8-RHOAR latest]} {perl openshift [5.16 5.20 5.24 5.26 latest]} {php openshift [5.5 5.6 7.0 7.1 latest]} {python openshift [2.7 3.3 3.4 3.5 3.6 latest]} {ruby openshift [2.0 2.2 2.3 2.4 2.5 latest]} {wildfly openshift [10.0 10.1 11.0 12.0 13.0 8.1 9.0 latest]}]
panic: cannot parse '200uu': unable to parse quantity's suffix

goroutine 1 [running]:
github.com/redhat-developer/odo/vendor/k8s.io/apimachinery/pkg/api/resource.MustParse(0x7ffe0ce3001e, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/redhat/git-srcs/src/github.com/redhat-developer/odo/vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go:133 +0x249
github.com/redhat-developer/odo/pkg/util.FetchResourceQunatity(0x1836baf, 0x6, 0x7ffe0ce3001e, 0x5, 0x0, 0x0, 0x0, 0x0, 0x4)
	/home/redhat/git-srcs/src/github.com/redhat-developer/odo/pkg/util/util.go:272 +0x306
github.com/redhat-developer/odo/cmd.glob..func22(0x2aabbe0, 0xc000b16150, 0x1, 0x7)
	/home/redhat/git-srcs/src/github.com/redhat-developer/odo/cmd/create.go:162 +0x748
github.com/redhat-developer/odo/vendor/github.com/spf13/cobra.(*Command).execute(0x2aabbe0, 0xc000b160e0, 0x7, 0x7, 0x2aabbe0, 0xc000b160e0)
	/home/redhat/git-srcs/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:702 +0x2d3
github.com/redhat-developer/odo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2aadbc0, 0x8, 0xc000000300, 0x145ae8e)
	/home/redhat/git-srcs/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:783 +0x2dc
github.com/redhat-developer/odo/vendor/github.com/spf13/cobra.(*Command).Execute(0x2aadbc0, 0x1935ac0, 0xc000122720)
	/home/redhat/git-srcs/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:736 +0x2b
github.com/redhat-developer/odo/cmd.Execute()
	/home/redhat/git-srcs/src/github.com/redhat-developer/odo/cmd/root.go:108 +0xae
main.main()
	/home/redhat/git-srcs/src/github.com/redhat-developer/odo/main.go:40 +0x1d1
```

Reply:
I tried the command
```
odo create nodejs --max-memory 300 --min-memory 100
```
The pod creation failed but the command
```
odo create nodejs --max-memory 300M --min-memory 100
```
seems to work. Weird that passing a unit for one worked for me.